xarray/backends/netCDF4_.py

```diff
@@ -399,7 +404,9 @@ def prepare_variable(self, name, variable, check_encoding=False,
             # OrderedDict as the input to setncatts
             nc4_var.setncattr(k, v)
 
-        return nc4_var, variable.data
+        target = NetCDF4ArrayWrapper(name, self)
+
+        return target, variable.data
```

Comment:
@jhamman 
This is too late, but I think `nc4_var` is never used. Is it correct?

Reply:
@fujiisoup - it is used in line 405 (`nc4_var.setncattr(k, v)`).